test/framework/src/main/java/org/elasticsearch/transport/nio/channel/TcpReadContext.java

```diff
@@ -75,7 +75,11 @@ public int read() throws IOException {
 
             try {
                 BytesReference messageWithoutHeader = message.slice(6, message.length() - 6);
-                handler.handleMessage(messageWithoutHeader, channel, channel.getProfile(), messageWithoutHeader.length());
+
+                // A message length of 6 bytes it is just a ping. Ignore for now.
```

Comment:
I added a comment next to the usage of frame decoder to indicate that it would thrown an exception if the message turned out to be invalid.